src/qibolab/pulses.py

```diff
@@ -1400,7 +1402,60 @@ def modulated_waveforms(self):  #  -> tuple[Waveform, Waveform]:
 
     @property
     def serial(self):
-        return f"FluxPulse({self.start}, {self.duration}, {format(self.amplitude, '.6f').rstrip('0').rstrip('.')}, {self.shape}, {self.channel}, {self.qubit})"
+        if isinstance(self.qubit, Coupler):
+            return f"Coupler_FluxPulse({self.start}, {self.duration}, {format(self.amplitude, '.6f').rstrip('0').rstrip('.')}, {self.shape}, {self.channel}, {self.qubit})"
+        else:
+            return f"FluxPulse({self.start}, {self.duration}, {format(self.amplitude, '.6f').rstrip('0').rstrip('.')}, {self.shape}, {self.channel}, {self.qubit})"
```

Comment:
You can even change this to
```suggestion
            return f"{self.__class__.__name__}({self.start}, {self.duration}, {format(self.amplitude, '.6f').rstrip('0').rstrip('.')}, {self.shape}, {self.channel}, {self.qubit})"
```
so that you won't even have to overwrite the `serial` in the `CouplerFluxPulse` below.

You can also play a similar game with the `type` in `__init__` but for that you would have to define a new class variable.

Reply:
Great, now it's just a single line the `CouplerFluxPulse`.